platforms/quorum/charts/quorum-node/templates/node-hooks-pre-install.yaml

```diff
@@ -31,12 +31,23 @@ spec:
           image: {{ .Values.image.hooks.repository }}:{{ .Values.image.hooks.tag }}
           securityContext:
             runAsUser: 0
-{{- if eq .Values.global.vault.type "hashicorp" }}
+{{- if or (eq .Values.global.vault.type "hashicorp") (and (.Values.global.cluster.cloudNativeServices) (eq .Values.global.cluster.provider "aws")) }}
```

Comment:
Conditions are too complex.
- check first if cloudNativeServices = true, then check if provider = aws or azure etc
- If cloudNativeServices = false, only the vault.type = hashicorp or Kubernetes
- There cannot be a situation where cloudNativeServices=true and vault.type=hashicorp, if cloudNativeServices = true, vault.type is ignored. Add all cloud KMS related keys under global.cluster itself